File: cpp/day-2/part-1.cpp
#include <regex>

#include "findall.hpp"
#include "iostream"
#include "read.hpp"

using namespace std;

auto main() -> int {
  auto filename = "./input/day-2.txt";
  auto lines = read(filename);
  regex pattern(
      "(?:1[3-9]|[2-9]\\d) r|(?:1[4-9]|[2-9]\\d) g|(?:1[5-9]|[2-9]\\d) b");
  int sum{0};

  for (unsigned short i = 1; const auto line : lines) {
    auto matches = findall(line, pattern);
    sum = !matches.size() ? sum + i : sum;
    i++;
  };

  cout << sum << endl;

  return 0;
}

File: python/day-2/part_1.py
from typing import List
import re


def solve(lines: List[str]) -> None:
    matches = [
        re.findall(
            r'(?:1[3-9]|[2-9]\d) r|(?:1[4-9]|[2-9]\d) g|(?:1[5-9]|[2-9]\d) b', line
        )
        for line in lines
    ]
    total = sum([i + 1 for i, match in enumerate(matches) if not bool(len(match))])

    print(total)
